main.c

```diff
@@ -1657,6 +1666,54 @@ static struct PyModuleDef gmp_module = {
 };
 
 
+#define max_alloc 256
```

Comment:
Avoid the inability to distinguish between macros and variable. 
```suggestion
#define MAX_ALLOC_SIZE 256
```
Or
```suggestion
#define MAX_HEAP_SIZE 256
```

Reply:
This is just a draft.  Hardly this dumb structure to track allocations will survive in a final version.  So, please hold on for nitpicks, until this will be marked as ready to review.